fileshare/fileshare_process/fileshare_process_manager.go

```diff
@@ -0,0 +1,76 @@
+package fileshare_process
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	childprocess "github.com/NordSecurity/nordvpn-linux/child_process"
+	"github.com/NordSecurity/nordvpn-linux/fileshare/pb"
+	"github.com/NordSecurity/nordvpn-linux/internal"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+var FileshareURL = fmt.Sprintf("%s://%s", internal.Proto, internal.FileshareSocket)
+
+type FileshareProcessClient struct {
+	mu sync.Mutex
+}
+
+func NewFileshareProcessClient() *FileshareProcessClient {
+	return &FileshareProcessClient{}
+}
+
+func getFileshareClient() (pb.FileshareClient, *grpc.ClientConn, error) {
+	fileshareConn, err := grpc.Dial(
+		FileshareURL,
+		grpc.WithTransportCredentials(insecure.NewCredentials()))
+
+	if err != nil {
+		return nil, nil, fmt.Errorf("connecting to fileshare daemon: %w", err)
+	}
+
+	client := pb.NewFileshareClient(fileshareConn)
+	return client, fileshareConn, nil
+}
+
+func (f *FileshareProcessClient) Ping(nowait bool) error {
+	f.mu.Lock()
+	defer f.mu.Unlock()
+
+	client, clientConn, err := getFileshareClient()
+	defer clientConn.Close()
```

Comment:
shouldn't defer be after error checking? otherwise it might crash if clientConn is nil.